### PR TITLE
Feature/script configurable

### PR DESCRIPTION
**Pull request by nicktming**
For this issue: #95 

add scripts to make it configurable for users when add/removing a dir.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ Kubernetes v1.12+.
 In this setup, the directory `/opt/local-path-provisioner` will be used across all the nodes as the path for provisioning (a.k.a, store the persistent volume data). The provisioner will be installed in `local-path-storage` namespace by default.
 
 ```
-kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/master/deploy/local-path-storage.yaml
+kuebctl apply -f deploy/local-path-storage.yaml
```
Comment (yasker) [inline README.md:29]: Hmm, we cannot do this. This will stop the one-line installation of Local Path Provisioner.

Better have the script in the config map by default.

File: provisioner.go [hunk 2]
```diff
@@ -344,6 +344,11 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
 							ReadOnly:  false,
 							MountPath: "/data/",
 						},
+						{
+							Name:      "script",
+							ReadOnly:  false,
+							MountPath: "/tmp",
```
Comment (yasker) [inline provisioner.go:350]: Don't use `/tmp`. It's a part of the helper pod anyway.

File: provisioner.go [hunk 3]
```diff
@@ -358,6 +363,26 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
...
+				{
+					Name: "script",
+					VolumeSource: v1.VolumeSource{
+						ConfigMap: &v1.ConfigMapVolumeSource{
+							LocalObjectReference: v1.LocalObjectReference{
+								Name: "script",
+							},
+							Items: []v1.KeyToPath{
+								{
+									Key:  "add.sh",
```
Comment (yasker) [inline provisioner.go:375]: We can call the script `setup`, `teardown`.

And it doesn't even need to be `.sh` since the interpreter should be specified at the first line of the script?

File: provisioner.go [hunk 4]
```diff
@@ -323,7 +322,8 @@ func (p *LocalPathProvisioner) createHelperPod(action ActionType, cmdsForPath []
 	hostPathType := v1.HostPathDirectoryOrCreate
 	helperPod := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: string(action) + "-" + name,
+			Name:      string(action) + "-" + name,
+			Namespace: p.namespace,
```
Comment (yasker) [inline provisioner.go:326]: This line should be a separate commit since normally we create the helper pod in the default namespace. Btw, why do we need this?

File: deploy/cm/add.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
```
Comment (yasker) [inline deploy/cm/add.sh:1]: how about just `#!/bin/bash`?

File: provisioner.go (hunk 3 above)
Comment (nicktming) [reply provisioner.go]: i have changed add.sh&del.sh to setup&teardown

File: provisioner.go (hunk 2 above)
Comment (nicktming) [reply provisioner.go]: i chaned /tmp to /script which is not a part of the helper pod.

File: README.md (hunk 1 above)
Comment (nicktming) [reply README.md]: i removed cm dir and put the scripts into deploy/local-path-storage.yam with a script configmap by default.

File: deploy/cm/add.sh (hunk 5 above)
Comment (nicktming) [reply deploy/cm/add.sh]: yes, i chanaged it to #!/bin/sh.

File: provisioner.go (hunk 4 above)
Comment (nicktming) [reply provisioner.go:326]: Since the helper pod needs to use the newly created configmap (script) which is in the same namespace as provisioner, then it is necessary to create/delete the helper pod with the provisioner's namespace.

**Comment by yasker** [comment]
The code part looks good except for I think we can combine the configs.

A few other things:
1. Can you squash everything into one commit? Since it's one feature and is not complex enough to have functionally separated commits. You can use `git rebase -i` for that.
2. Can you update the documentation [Configuration section](https://github.com/rancher/local-path-provisioner#configuration), as well as [the example config](https://github.com/rancher/local-path-provisioner/blob/master/deploy/example-config.yaml)? Those can be in a separate commit or merged with the first one.
3. We need to update the [Helm Chart](https://github.com/rancher/local-path-provisioner/tree/master/deploy/chart). This should be included in the first commit.

File: deploy/local-path-storage.yaml [hunk 6]
```diff
@@ -103,3 +103,19 @@ data:
                 ]
         }
 
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: script
```
[the remark this answers is not in the document]
Comment (nicktming) [reply deploy/local-path-storage.yaml]: ok, it is done.

**Comment by nicktming** [comment]
> The code part looks good except for I think we can combine the configs.
> 
> A few other things:
> 
>     1. Can you squash everything into one commit? Since it's one feature and is not complex enough to have functionally separated commits. You can use `git rebase -i` for that.
>  ok, it is done.

>     2. Can you update the documentation [Configuration section](https://github.com/rancher/local-path-provisioner#configuration), as well as [the example config](https://github.com/rancher/local-path-provisioner/blob/master/deploy/example-config.yaml)? Those can be in a separate commit or merged with the first one.
>  ok, it is done. i updated the documentation in readme and example config, i will commit if it needs more in the future.

**Comment by yasker** [comment]
Hi @nicktming , how about the chart?

**Comment by nicktming** [comment]
> Hi @nicktming , how about the chart?

ok, it is done.

**Comment by yasker** [comment]
Merged, thanks @nicktming !
